ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/SQSTest.kt

```diff
@@ -0,0 +1,261 @@
+                    // Make sure it shows a sent message
+                    findByXpath("//div[contains(@accessiblename, 'Sent message ID')]")
+                }
+                step("Add 10 messages to the queue so poll will be guaranteed to work") {
+                    client.sendMessageBatch {
+                        it.queueUrl(queueUrl).entries(
+                            (1..10).map { num ->
+                                SendMessageBatchRequestEntry.builder().messageBody("bmessage$num").id(num.toString()).build()
+                            }
+                        )
+                    }
+                }
+                openPollMessagePane(queueName)
+                step("View messages") {
+                    findAndClick("//div[@accessiblename='View Messages' and @class='JButton']")
+                    // Wait for the table to be populated (It's very fast for small queues)
+                    Thread.sleep(1000)
+                    find<JTreeFixture>(byXpath("//div[@class='TableView']")).findAllText().any { it.text.contains("bmessage") }
+                }
+            }
+            closeToolWindowTab()
+            step("FIFO queue") {
+                openSendMessagePane(fifoQueueName)
+                step("Send a message and validate it is sent") {
+                    // fill the message box
+                    fillSingleJBTextArea("message")
+                    // Fill the rest of the fields (deduplication and group ids)
+                    fillAllJBTextFields("message")
+                    findAndClick("//div[@text='Send']")
+                    // Make sure it shows a sent message
+                    findByXpath("//div[contains(@accessiblename, 'Sent message ID')]")
+                }
+            }
+            closeToolWindowTab()
+            step("edit queue settings") {
+                awsExplorer {
+                    openExplorerActionMenu(sqsNodeLabel, queueName)
+                }
+                findAndClick("//div[@text='$editQueueAttributesAction']")
+                // Make sure the dialog opens, then close it
+                find<NewProjectWizardDialog>(DialogFixture.byTitle(editQueueAttributesTitle)).close()
+            }
+            step("Purge queue") {
+                awsExplorer {
+                    openExplorerActionMenu(sqsNodeLabel, queueName)
+                }
+                findAndClick("//div[@text='$purgeQueueText']")
+                pressYes()
+                val toast = findToast()
+                assertThat(toast.hasText { it.text.contains("Started purging queue") })
+                awsExplorer {
+                    openExplorerActionMenu(sqsNodeLabel, queueName)
+                }
+                findAndClick("//div[@text='$purgeQueueText']")
+                pressYes()
+                val errorToast = findToast()
+                assertThat(errorToast.hasText { it.text.contains("Purge queue request already in progress for queue") })
```

Comment:
This feels like it might be a bit fragile and lead to a timing issue ...

Reply:
It's a 60 second period, so I don't think it will be that fragile. I like having it, because this is the only way to test it in a reasonable way.